conda/resolve.py

```diff
@@ -977,6 +983,10 @@ def generate_version_metrics(self, C, specs, include0=False):
 
         sdict = {}  # Dict[package_name, PackageRecord]
 
+        prerelease_penalty = 0
+        if context.prerelease_behavior is not PrereleaseBehavior.ALLOW:
+            prerelease_penalty = 100
```

Comment:
Is there any specific reason we use `100` as penalty?

Reply:
No. It just needs to be a large finite number to ensure that the prerelease solutions go last.

I think that 100 is actually probably not large enough. Probably should just make it 1000000 or something.